pkg/ibgu/ibgu_test.go

```diff
@@ -432,3 +440,200 @@ func generateIbgu() *v1alpha1.ImageBasedGroupUpgrade {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: v1alpha1.ImageBasedGroupUpgradeSpec{},
+		}
+	}
+
+	testCases := []struct {
+		ibguName            string
+		ibguNamespace       string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		client              bool
+	}{
+		{
+			ibguName:            "test1",
+			ibguNamespace:       "test-namespace",
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			client:              true,
+		},
+		{
+			ibguName:            "test2",
+			ibguNamespace:       "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "ibgu object test2 does not exist in namespace test-namespace",
+			client:              true,
+		},
+		{
+			ibguName:            "",
+			ibguNamespace:       "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "ibgu 'name' cannot be empty",
+			client:              true,
+		},
+		{
+			ibguName:            "test3",
+			ibguNamespace:       "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "ibgu 'namespace' cannot be empty",
+			client:              true,
+		},
+		{
+			ibguName:            "test3",
+			ibguNamespace:       "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "ibgu 'apiClient' cannot be empty",
+			client:              false,
+		},
+	}
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testIbgu := generateibgu(testCase.ibguName, testCase.ibguNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIbgu)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		// Test the Pull method
+		builderResult, err := PullIbgu(testSettings, testIbgu.Name, testIbgu.Namespace)
+
+		// Check the error
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			// Check the error message
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testIbgu.Name, builderResult.Object.Name)
+			assert.Equal(t, testIbgu.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestIbguDeleteAndWait(t *testing.T) {
+	testCases := []struct {
+		testibgu      *IbguBuilder
+		expectedError error
+	}{
+		{
+			testibgu:      buildValidIbguTestBuilder(buildTestClientWithDummyIbguObject()),
+			expectedError: nil,
+		},
+		{
+			testibgu:      buildValidIbguTestBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("ibgu cannot be deleted because it does not exist"),
```

Comment:
ibgu delete doesn't return an error in this case

Reply:
Removed this test- thanks.